### ISSUE TITLE
Why there is no detach() after the generator?

### ISSUE THREAD

**Issue by ToTheBeginning** [znxlwm/pytorch-generative-model-collections#4]
I see many code like this:`fake_data = G(noise).detach()`

But in your most code, there is no detach, so is it just ok to omit it ?

**Comment by aidiary** [comment]
https://github.com/pytorch/examples/issues/116
